client/src/AppRoot.tsx

```diff
@@ -36,49 +37,72 @@ const AppRoot = () => {
     history.push(routerPaths.root)
   }
 
-  const addElection = (newElection: MailElection) => {
-    const newElections = (mailElections || []).concat([newElection])
-    setMailElections(newElections)
-  }
+  const updateMailElections = useCallback(async () => {
+    try {
+      const { mailElections } = await getMailElections()
+      setMailElections(mailElections)
+    } catch (error) {
+      history.push(routerPaths.root)
+      window.location.reload(false)
+      console.error('getAndSetMailElections failed', error) // eslint-disable-line no-console
+    }
+  }, [history])
+
+  useEffect(() => {
+    updateMailElections()
+  }, [
+    setMailElections,
+    updateMailElections,
+    user,
+  ])
+
+  const updateVoters = useCallback(async () => {
+    mailElections?.forEach(async me => {
+      const { voters: newVoters } = await getVoters({ electionId: me.id })
```

Comment:
why download voters for all elections? This could be quite wasteful.

Voter list should probably be loaded only when the voters tab is viewed for a single election.